### PR TITLE
Enable nullability in RelatedCurrencyManager

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in RelatedCurrencyManager.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/9897)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/RelatedCurrencyManager.cs [hunk 1]
```diff
@@ -173,25 +172,25 @@ private void ParentManager_CurrentItemChanged(object sender, EventArgs e)
...
                 {
-                    IgnoreItemChangedTable.Add(curManager);
-                    curManager.CancelCurrentEdit();
+                    IgnoreItemChangedTable.Add(currencyManager);
+                    currencyManager.CancelCurrentEdit();
                 }
                 finally
                 {
-                    if (IgnoreItemChangedTable.Contains(curManager))
+                    if (IgnoreItemChangedTable.Contains(currencyManager))
```
Comment (halgab) [inline src/System.Windows.Forms/src/System/Windows/Forms/RelatedCurrencyManager.cs:183]: I guess this line is redundant?

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/RelatedCurrencyManager.cs:183]: Based on the long comment above, I don't think so.

Comment (halgab) [reply src/System.Windows.Forms/src/System/Windows/Forms/RelatedCurrencyManager.cs:183]: I'm not sure I get how it addresses the need to call Contains before Remove, but it does indicate this code is tricky, so I guess staying on the safe side and not changing anything is not a bad idea...

**Review by lonitra — APPROVED** [review]
LGTM

**Pull request merged**